wp-content/civi-extensions/goonjcustom/api/v3/Goonjcustom/CollectionCampCron.php

```diff
@@ -62,85 +62,93 @@ function civicrm_api3_goonjcustom_collection_camp_cron($params) {
+        ->addWhere('id', '=', $recipientId)
+        ->execute()->single();
+
+      $contactName = $contact['display_name'];
+
+      $fromEmail = OptionValue::get(FALSE)
+        ->addSelect('label')
+        ->addWhere('option_group_id:name', '=', 'from_email_address')
+        ->addWhere('is_default', '=', TRUE)
+        ->execute()->single();
+
+      // Send email if the end date is today or earlier.
+      if ($endDateFormatted <= $todayFormatted) {
+        $mailParams = [
+          'subject' => 'Volunteer Feedback Form',
+          'from' => $fromEmail['label'],
+          'toEmail' => $contactEmailId,
+          'replyTo' => $fromEmail['label'],
+          'html' => goonjcustom_collection_camp_volunteer_feedback_email_html($organizingContactName, $collectionCampId),
+        ];
+        $result = CRM_Utils_Mail::send($mailParams);
+      }
+
+      // Process activities.
+      $activities = Activity::get(FALSE)
+        ->addSelect('id')
+        ->addWhere('Material_Contribution.Collection_Camp', '=', $collectionCampId)
+        ->execute();
+
+      $contributorCount = count($activities);
+
+      $results = EckEntity::update('Collection_Camp', FALSE)
+        ->addValue('Camp_Outcome.Number_of_Contributors', $contributorCount)
+        ->addWhere('id', '=', $collectionCampId)
+        ->execute();
+
+      // Send completion notification.
+      if ($endDateFormatted <= $todayFormatted) {
+        $mailParams = [
+          'subject' => 'Collections Completion Notification',
+          'from' => 'urban.ops@goonj.org',
+          'toEmail' => $emailId,
+          'replyTo' => 'urban.ops@goonj.org',
+          'html' => goonjcustom_collection_camp_email_html($contactName, $collectionCampId, $recipientId, $collectionCampGoonjOffice),
+        ];
+        $result = CRM_Utils_Mail::send($mailParams);
+      }
     }
-
-    $activities = Activity::get(FALSE)
-      ->addSelect('id')
-      ->addWhere('Material_Contribution.Collection_Camp', '=', $collectionCampId)
-      ->execute();
-
-    $contributorCount = count($activities);
-
-    $results = EckEntity::update('Collection_Camp', FALSE)
-      ->addValue('Camp_Outcome.Number_of_Contributors', $contributorCount)
-      ->addWhere('id', '=', $collectionCampId)
-      ->execute();
-
-    // Only send the email if the end date is lower than today.
-    if ($endDateFormatted <= $todayFormatted) {
-      $mailParams = [
-        'subject' => 'Collections Completion Notification',
-        'from' => 'urban.ops@goonj.org',
-        'toEmail' => $emailId,
-        'replyTo' => 'urban.ops@goonj.org',
-        'html' => goonjcustom_collection_camp_email_html($contactName, $collectionCampId, $recipientId, $collectionCampGoonjOffice),
-        // 'messageTemplateID' => 76, // Uncomment if using a message template
-      ];
-      $result = CRM_Utils_Mail::send($mailParams);
+    catch (Exception $e) {
+      error_log("Error processing camp ID $collectionCampId: " . $e->getMessage());
```

Comment:
Can we log all the CRM related exception in CiviCRM logs instead of WordPress logs?

```suggestion
      \Civi::log()->info("Error processing camp ID $collectionCampId: " . $e->getMessage());
```